py/desispec/scripts/submit_night.py

```diff
@@ -371,7 +390,7 @@ def submit_night(night, proc_obstypes=None, z_submit_types=None, queue='realtime
             flats.append(prow)
         elif curtype == 'arc' and calibjobs['psfnight'] is None:
             arcs.append(prow)
-        elif curtype == 'science' and prow['LASTSTEP'] != 'skysub':
+        elif curtype == 'science' and (prow['LASTSTEP'] != 'skysub' or use_tilenight):
```

Comment:
@akremin prow corresponds to the new exposure, since it is defined in line 368:
```
prow = define_and_assign_dependency(prow, calibjobs)
```
and that statement is executed regardless of whether `use_tilenight==True` or not. So I think I'm appending a new exposure here in the case `use_tilenight==True`. 

Please let me know if this makes sense, and I will mark this conversation as resolved.

Reply:
@marcelo-alvarez can you clarify: on line 368 (line numbers from this PR) `prow` is created regardless of `use_tilenight` True or False., but where is the equivalent logic for the block under line 369:
```
if (not use_tilenight) or erow['OBSTYPE'] != 'science':
    ...
    prow = create_and_submit(prow,  ...)
    ....
    ptable.add_row(prow)
    ...
```
for the case `use_tilenight==True and erow['OBSTYPE'] == 'science'`?

Reply:
@sbailey when `use_tilenight=True` there are no processing steps for individual science exposures, so there are no science exposure `prow`s to append, and hence there is no equivalent logic for the block under line 369, the purpose of which is to create and submit per-exposure steps and append them to the processing table.

Rather, the `prow` created at line 368 (for the version you reference above) 
```
        prow = define_and_assign_dependency(prow, calibjobs)
```
when `use_tilenight=True` is only used to append to the list of science exposures, i.e. 
```
        elif curtype == 'science' and (prow['LASTSTEP'] != 'skysub' or use_tilenight):
            sciences.append(prow)
```